File: beginner/1021.c
#include <stdio.h>
#include <math.h>

int main()
{	
	double valor; 
	int restoMoeda;
	scanf ("%lf", &valor);
	valor *= 1000;
	printf("NOTAS:\n");
	printf("%d nota(s) de R$ 100.00\n",(int)valor / 100000);
	printf("%d nota(s) de R$ 50.00\n", ((int)valor % 100000) / 50000);
	printf("%d nota(s) de R$ 20.00\n", (((int)valor % 100000) % 50000) / 20000);
	printf("%d nota(s) de R$ 10.00\n", ((((int)valor % 100000) % 50000) % 20000) / 10000);
	printf("%d nota(s) de R$ 5.00\n",  (((((int)valor % 100000) % 50000) % 20000) % 10000) / 5000);	
	printf("%d nota(s) de R$ 2.00\n",  ((((((int)valor % 100000) % 50000) % 20000) % 10000) % 5000) / 2000);	

	restoMoeda = (((((((int)valor % 100000) % 50000) % 20000) % 10000) % 5000) % 2000);

	printf("MOEDAS:\n");
	printf("%d moeda(s) de R$ 1.00\n", restoMoeda / 1000);
	printf("%d moeda(s) de R$ 0.50\n", (restoMoeda % 1000) / 500);
	printf("%d moeda(s) de R$ 0.25\n", ((restoMoeda % 1000) % 500) / 250);
	printf("%d moeda(s) de R$ 0.10\n", (((restoMoeda % 1000) % 500) % 250) / 100);
	printf("%d moeda(s) de R$ 0.05\n", ((((restoMoeda % 1000) % 500) % 250) % 100) / 50);	
	printf("%d moeda(s) de R$ 0.01\n", (((((restoMoeda % 1000) % 500) % 250) % 100) % 50) / 10);

File: beginner/1021.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var total = parseFloat(lines.shift());

console.log('NOTAS:');

console.log(parseInt(total / 100) + ' nota(s) de R$ 100.00');
total = total % 100;
console.log(parseInt(total/50) + ' nota(s) de R$ 50.00');
total %= 50;
console.log(parseInt(total/20) + ' nota(s) de R$ 20.00');
total %= 20;
console.log(parseInt(total/10) + ' nota(s) de R$ 10.00');
total %= 10;
console.log(parseInt(total/5) + ' nota(s) de R$ 5.00');
total %= 5;
console.log(parseInt(total/2) + ' nota(s) de R$ 2.00');
total %= 2;

console.log('MOEDAS:')
console.log(parseInt(total/1) + ' moeda(s) de R$ 1.00');
total %= 1;
total *= 100;
console.log(parseInt(total/50) + ' moeda(s) de R$ 0.50');
total %= 50;
console.log(parseInt(total/25) + ' moeda(s) de R$ 0.25');
total %= 25;
console.log(parseInt(total/10) + ' moeda(s) de R$ 0.10');
total %= 10;
console.log(parseInt(total/5) + ' moeda(s) de R$ 0.05');
total %= 5;
total += 0.00001;
console.log(parseInt(total) + ' moeda(s) de R$ 0.01');
